### PR TITLE
Start Flake8 checking of Numba source

### PR DESCRIPTION
**Pull request by seibert**
A long-standing goal for Numba is enforcing code style using a tool like flake8.  To avoid disrupting the code history by trying to bring the entire code base up to compliance at once, this PR instead introduces a flake8 task into the Azure CI testing, but excludes all currently failing source files (which is most of them).

With this PR, the new policy will be:
1. New files need to be compliant.
2. When doing substantial edits to a file, remove it from the exclude list and fix the style errors.  Do this as a standalone commit to make the PR easier to review.
3. Reviewers should check that the above two rules are followed.

In the Numba 0.42 dev cycle, we'll be reviewing files for other issues (code that doesn't follow current Numba best practice, etc), at which time we'll also fix the style issues.  For now, reformat-only PRs are not desired.

### CODE REVIEW

**Comment by seibert** [comment]
Also, the flake8 configuration is taken from Dask's [setup.cfg](https://github.com/dask/dask/blob/master/setup.cfg#L1-L19).  Open to suggestions about how to tweak it, if people have opinions.

**Comment by stuartarchibald** [comment]
Seems like `numba/rewrites/ir_print.py` needs adding to the ignore list as merge to master pulls in non-compliant version. Else change is good, thanks.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+    E20,   # Extra space in brackets
+    E231,E241,  # Multiple spaces around ","
+    E26,   # Comments
+    E4,    # Import formatting
+    E721,  # Comparing types instead of isinstance
+    E731,  # Assigning lambda expression
+    E741,  # Ambiguous variable names
+    W503,  # line break before binary operator
+    W504,  # line break after binary operator
+    F811,  # redefinition of unused 'loop' from line 10
```
Comment (sklam) [inline .flake8]: adjust comment to redefinition of unused variables?

Comment (seibert) [reply .flake8]: Actually, it seems that F811 doesn't fail any additional files, so I'm going to drop it from the ignore list.

**Comment by seibert** [comment]
These error codes don't seem to flag any additional files, so I re-enabled them:
```
    E4,    # Import formatting
    E721,  # Comparing types instead of isinstance
    F811,  # redefinition of unused name
```

**Comment by seibert** [comment]
btw, this failed Azure because one of the jobs experienced a random 502 gateway error.  The job with the flake8 check did pass.

**Comment by stuartarchibald** [comment]
Thanks @seibert, I checked the Azure failure/pass state too, looks fine and correct as noted. Happy for this to be merged as a start to a long process. Thanks for doing the set up.

**Pull request merged**